src/stories/Dialog.stories.tsx

```diff
@@ -1,9 +1,57 @@
 import React, {useState, useRef, useCallback} from 'react'
 import {Meta} from '@storybook/react'
 
-import {BaseStyles, ThemeProvider, Box} from '..'
+import {BaseStyles, ThemeProvider, Box, Autocomplete, AutocompleteInputProps, SSRProvider, theme} from '..'
 import {Button} from '../Button'
 import {Dialog, DialogProps, DialogWidth, DialogHeight} from '../Dialog/Dialog'
+import {AutocompleteMenuInternalProps} from '../Autocomplete/AutocompleteMenu'
+import {ItemProps} from '../deprecated/ActionList'
+import {MandateProps} from '../utils/types'
+
+const mockItems = [
+  {text: 'zero', id: 0},
+  {text: 'one', id: 1},
+  {text: 'two', id: 2},
+  {text: 'three', id: 3},
+  {text: 'four', id: 4},
+  {text: 'five', id: 5},
+  {text: 'six', id: 6},
+  {text: 'seven', id: 7},
+  {text: 'twenty', id: 20},
+  {text: 'twentyone', id: 21},
+]
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AutocompleteItemProps<T = Record<string, any>> = MandateProps<ItemProps, 'id'> & {metadata?: T}
+
+const LabelledAutocomplete = <T extends AutocompleteItemProps>({
```

Comment:
Just a heads up, I have a slightly related PR that adds a Storybook example for rendering an Autocomplete in a Dialog: https://github.com/primer/react/pull/3374

Story in preview deployment: https://primer-59381bd591-13348165.drafts.github.io/storybook/?path=/story/components-forms-autocomplete--in-a-dialog